core/src/main/java/org/apache/calcite/util/format/postgresql/NumberFormatPattern.java

```diff
@@ -143,4 +165,40 @@ protected String trimLeadingZeros(String value) {
       return value;
     }
   }
+
+  @Override public ChronoUnitEnum getChronoUnit() {
+    return chronoUnit;
+  }
+
+  @Override protected int parseValue(final ParsePosition inputPosition, final String input,
+      Locale locale, boolean haveFillMode, boolean enforceLength) throws Exception {
+    int endIndex = inputPosition.getIndex();
+    for (; endIndex < input.length(); endIndex++) {
+      if (input.charAt(endIndex) < '0' || input.charAt(endIndex) > '9') {
+        break;
+      } else if (enforceLength && endIndex == inputPosition.getIndex() + preferredLength) {
+        break;
+      }
+    }
+
+    if (endIndex == inputPosition.getIndex()) {
+      throw new Exception();
+    }
+
+    int value = Integer.parseInt(input.substring(inputPosition.getIndex(), endIndex));
+    if (value < minValue || value > maxValue) {
+      throw new Exception();
```

Comment:
a better error message is needed

Reply:
Changed to a ParseException and added a message.